tests/testthat/test-tokens.R

```diff
@@ -400,21 +400,65 @@ test_that("tokens works for strange spaces (#796)", {
     expect_equal(ntoken(txt, remove_punct = FALSE, remove_separators = TRUE), c(text1 = 12))
     expect_equal(
         as.character(tokens(txt, remove_punct = TRUE, remove_separators = TRUE)),
-        c("space", "tab", "newline", "non-breakingspace", "em-space", "variationselector16")
+        c("space", "tab", "newline", "non", "breakingspace", "em", "space", "variationselector16")
     )
-    expect_equal(ntoken(txt, remove_punct = FALSE, remove_separators = FALSE), c(text1 = 22))
+    expect_equal(ntoken(txt, remove_punct = FALSE, remove_separators = FALSE), c(text1 = 26))
     expect_equal(
-        as.character(tokens(txt, remove_punct = FALSE, remove_separators = FALSE))[20:22],
+        as.character(tokens(txt, remove_punct = FALSE, remove_separators = FALSE))[24:26],
         c("variationselector16", " \uFE0F", ".")
     )
     expect_equal(
         ntoken(txt, remove_punct = TRUE, remove_separators = FALSE),
-        c(text1 = 14)
+        c(text1 = 22)
     )
     expect_equal(
-        as.character(tokens(txt, remove_punct = TRUE, remove_separators = FALSE))[13:14],
+        as.character(tokens(txt, remove_punct = TRUE, remove_separators = FALSE))[21:22],
         c("variationselector16", " \uFE0F")
     )
 })
 
+test_that("test tokens.tokens", {
+    
+chars <- c("a b c 12345 ! @ # $ % ^ & * ( ) _ + { } | : \' \" < > ? ! , . \t \n \u2028 \u00A0 \u2003 \uFE0F",
+           "#tag @user", "abc be-fg hi 100kg 2017", "https://github.com/kbenoit/quanteda", "a b c d e")
+toks1 <- as.tokens(stringi::stri_split_fixed(chars[1], ' '))
+toks2 <- as.tokens(stringi::stri_split_fixed(chars[2], ' '))
+toks3 <- as.tokens(stringi::stri_split_fixed(chars[3], ' '))
+toks4 <- as.tokens(stringi::stri_split_fixed(chars[4], ' '))
+toks5 <- as.tokens(stringi::stri_split_fixed(chars[5], ' '))
+
+expect_equal(tokens(chars[1], remove_numbers = TRUE),
+             tokens(toks1, remove_numbers = TRUE))
+
+expect_equal(tokens(chars[1], remove_punct = TRUE),
+             tokens(toks1, remove_punct = TRUE))
+
+expect_equal(tokens(chars[1], remove_separator = TRUE),
+             tokens(toks1, remove_separator = TRUE))
+
+expect_equal(tokens(chars[1], remove_symbols = TRUE),
+             tokens(toks1, remove_symbols = TRUE))
+
+expect_equal(tokens(chars[2], remove_punct = TRUE, remove_twitter = TRUE),
+             tokens(toks2, remove_punct = TRUE, remove_twitter = TRUE))
+
+expect_equal(tokens(chars[4], remove_url = TRUE),
+             tokens(toks4, remove_url = TRUE))
+
+expect_equal(tokens(chars[5], ngrams = 1:2),
+             tokens(toks5, ngrams = 1:2))
+
+expect_equal(tokens(chars[5], ngrams = 2, skip = 1:2),
+             tokens(toks5, ngrams = 2, skip = 1:2))
+
+# This fails because hyphnated words are concatenated in toks
```

Comment:
'-' is remove in the second case, because `stringi::stri_detect_regex('-', "^[\\p{P}]$") == TRUE`. It seems counter intuitive to make the effect of `remove_hyphens` conditional on `remove_punct`. 

It should be like
```r
tokens("One two-three four.", remove_hyphens = TRUE)
tokens from 1 document.
text1 :
[1] "One"   "two"   "-"     "three" "four"  "."    

> tokens("One two-three four.", remove_hyphens = FALSE)
tokens from 1 document.
text1 :
[1] "One"       "two-three" "four"   "."    
```